core/src/main/java/com/exadel/aem/toolkit/core/authoring/models/Autocomplete.java

```diff
@@ -0,0 +1,156 @@
+
+/**
+ * Represents the back-end part of the {@code Autocomplete} component for Granite UI dialogs. His Sling model
+ * is responsible for injecting and retrieving the properties of component's node.
+ */
+@Model(adaptables = SlingHttpServletRequest.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+@AemComponent(
+    path = "components/authoring/autocomplete",
+    title = "Autocomplete Coral 3",
+    resourceSuperType = "etoolbox-authoring-kit/components/authoring/base"
+)
+public class Autocomplete extends BaseModel {
+
+    @SlingObject
+    private SlingHttpServletRequest request;
+
+    @OSGiService
+    private OptionProviderService optionProvider;
+
+    @OSGiService
+    private ModelFactory modelFactory;
+
+    @ChildResource
+    @Named("items")
+    private List<Option> options = new ArrayList<>();
+
+    @ValueMapValue
+    private String placeholder;
+
+    @ValueMapValue
+    private boolean matchStartsWith;
+
+    @ValueMapValue
+    private String icon;
+
+    @ValueMapValue
+    private boolean disabled;
+
+    @ValueMapValue
+    private boolean invalid;
+
+    @ValueMapValue
+    private boolean loading;
+
+    @ValueMapValue
+    private boolean multiple;
+
+    /**
+     * Retrieves the list of {@link Option} objects
+     * @return List of {@link Option} objects
+     */
+    public List<Option> getOptions() {
+        return options;
+    }
+
+    /**
+     * Retrieves the specified placeholder value of the autocomplete
+     * @return String
+     */
+    public String getPlaceholder() {
+        return placeholder;
+    }
+
+    /**
+     * Retrieves the specification of the autocomplete if it is match text by starting with or contains
+     * @return true or false
+     */
+    public boolean isMatchStartsWith() {
+        return matchStartsWith;
+    }
+
+    /**
+     * Retrieves the specified icon value of the autocomplete
+     * @return String
+     */
+    public String getIcon() {
+        return icon;
+    }
+
+    /**
+     * Retrieves the specification of the autocomplete if it is disabled or not
+     * @return true or false
+     */
+    public boolean isDisabled() {
+        return disabled;
+    }
+
+    /**
+     * Retrieves the specification of the autocomplete if it is invalid or not
+     * @return true or false
+     */
+    public boolean isInvalid() {
+        return invalid;
+    }
+
+    /**
+     * Retrieves the specification of the autocomplete if it is show loading or not
+     * @return true or false
+     */
+    public boolean isLoading() {
+        return loading;
+    }
+
+    /**
+     * Retrieves the specification of the autocomplete if it is support multiple values or not
+     * @return true or false
+     */
+    public boolean isMultiple() {
+        return multiple;
+    }
+
+    /**
+     * Initializes this model per Sling Model standard
+     */
+    @PostConstruct
+    private void init() {
+        List<Option> optionsModels = optionProvider
```

Comment:
The Toolkit's own `@Children` annotation must be working instead of all these. Not 100% sure. But please report is it doesn't. Because that's exactly what it was designed for)